### DISCUSSION TITLE
Question: how to checkout remote repository?

### DISCUSSION THREAD

**Discussion by demurgos** [GitoxideLabs/gitoxide#358] [Q&A] [answered]
I am looking into using Gitoxide as a library to download the file tree at a given commit (or branch) from a remote repository.

Is this use-case already supported? Is there an example somewhere?

**Comment by Byron** [comment, marked as answer]
Thanks for your interest. I think doing that is possible, even though I have not done that before.

Here is the rough steps that I anticipate.

- start [cloning a pack from the remote](https://github.com/Byron/gitoxide/blob/2e1437cb0b5dc77f2317881767f71eaf9b009ebf/gitoxide-core/src/pack/receive.rs#L65), but declare it as shallow with a depth of one to receive only a single commit along with all objects associated with it. 
    - possibly use an object filter to not download blobs
- once the pack is written to disk and the index is generated, you can use the `git-odb` crate to access the objects within, which includes [breadth-first tree traversal](https://github.com/Byron/gitoxide/blob/2f2d856efe733d3cf81110c0e0607d2e7c40d968/git-traverse/src/tree/breadthfirst.rs#L60) in the `git-traverse` crate.

I hope that helps.
